library/src/scala/annotation/internal/AssignedNonLocally.scala

```diff
@@ -0,0 +1,7 @@
+package scala.annotation
+package internal
+
+/** An annotation to indicate that a private `var` was assigned with a prefix
+ *  other than the `this` type of its owner.
+ */
+class AssignedNonLocally() extends Annotation
```

Comment:
do these annotation classes have any consequences for library compatibility?

Reply:
no, it's internal compiler stuff only.